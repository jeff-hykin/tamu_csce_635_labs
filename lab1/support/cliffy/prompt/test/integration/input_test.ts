import { Input } from "../../input.ts";
import { snapshotTest } from "../../../testing/snapshot.ts";

await snapshotTest({
  name: "input prompt",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should enter some text": { stdin: ["foo bar", "\n"] },
  },
  async fn() {
    await Input.prompt({
      message: "Whats your name?",
      default: "foo",
    });
  },
});

await snapshotTest({
  name: "input prompt with suggestions",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should enable suggestions and list": { stdin: ["foo", "\n"] },
  },
  async fn() {
    await Input.prompt({
      message: "Whats your name?",
      default: "foo",
      suggestions: ["foo", "bar", "baz"],
      list: true,
    });
  },
});

await snapshotTest({
  name: "input prompt with prefix",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should change prefix": { stdin: ["bar", "\n"] },
  },
  async fn() {
    await Input.prompt({
      message: "Whats your name?",
      default: "foo",
      prefix: "PREFIX ",
    });
  },
});

await snapshotTest({
  name: "input prompt with writer set to stderr",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should enter some text": { stdin: ["foo bar", "\n"] },
  },
  async fn() {
    await Input.prompt({
      message: "Whats your name?",
      default: "foo",
      writer: Deno.stderr,
    });
  },
});

await snapshotTest({
  name: "input prompt without prefix",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should disable prefix": { stdin: ["bar", "\n"] },
  },
  async fn() {
    await Input.prompt({
      message: "Whats your name?",
      default: "foo",
      prefix: "",
    });
  },
});

await snapshotTest({
  name: "input prompt with no location flag",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should work without --location flag": { stdin: ["yes", "\n"] },
  },
  async fn() {
    await Input.prompt({
      message: "Works without --location?",
      default: "hope so",
    });
  },
});
