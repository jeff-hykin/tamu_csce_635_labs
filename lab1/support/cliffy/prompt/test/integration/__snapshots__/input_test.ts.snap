export const snapshot = {};

snapshot[`input prompt > should enter some text 1`] = `
stdout:
"? Whats your name? (foo) › \\x1b[28G\\x1b[G\\x1b[0J
? Whats your name? (foo) › foo bar\\x1b[35G\\x1b[G\\x1b[0J
? Whats your name? (foo) › foo bar
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`input prompt with suggestions > should enable suggestions and list 1`] = `
stdout:
"? Whats your name? (foo) › 
  foo
  bar
  baz\\x1b[3A\\x1b[28G\\x1b[G\\x1b[0J
? Whats your name? (foo) › foo
❯ foo\\x1b[1A\\x1b[31G\\x1b[G\\x1b[0J
? Whats your name? (foo) › foo
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`input prompt with prefix > should change prefix 1`] = `
stdout:
"PREFIX Whats your name? (foo) › \\x1b[33G\\x1b[G\\x1b[0J
PREFIX Whats your name? (foo) › bar\\x1b[36G\\x1b[G\\x1b[0J
PREFIX Whats your name? (foo) › bar
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`input prompt with writer set to stderr > should enter some text 1`] = `
stdout:
""
stderr:
"? Whats your name? (foo) › \\x1b[28G\\x1b[G\\x1b[0J
? Whats your name? (foo) › foo bar\\x1b[35G\\x1b[G\\x1b[0J
? Whats your name? (foo) › foo bar
\\x1b[?25h\\x1b[?25h"
`;

snapshot[`input prompt without prefix > should disable prefix 1`] = `
stdout:
"Whats your name? (foo) › \\x1b[26G\\x1b[G\\x1b[0J
Whats your name? (foo) › bar\\x1b[29G\\x1b[G\\x1b[0J
Whats your name? (foo) › bar
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`input prompt with no location flag > should work without --location flag 1`] = `
stdout:
"? Works without --location? (hope so) › \\x1b[41G\\x1b[G\\x1b[0J
? Works without --location? (hope so) › yes\\x1b[44G\\x1b[G\\x1b[0J
? Works without --location? (hope so) › yes
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
